java/core/src/main/java/sleeper/core/range/Region.java

```diff
@@ -62,15 +62,37 @@ public Range getRange(String fieldName) {
     }
 
     /**
-     * Returns the ranges in this region. Note that these are returned in no
-     * particular order.
+     * Returns the ranges in this region unordered.
+     *
+     * <strong>Note: these ranges are returned in no
+     * particular order.</strong>
      *
      * @return a List of Ranges in no particular order.
+     * @see    Region#getRangesOrdered(Schema)
      */
-    public List<Range> getRanges() {
+    public List<Range> getRangesUnordered() {
```

Comment:
If we change the return type to Collection, we can avoid copying the list. If we do that we can use `Collections.unmodifiableCollection` to make it immutable.